File: android/src/main/java/com/rnfs2/RNFSFileTransformer.java
package com.rnfs2;

public class RNFSFileTransformer {
  public interface FileTransformer {
    public byte[] onWriteFile(byte[] data);
    public byte[] onReadFile(byte[] data);
  }

  public static RNFSFileTransformer.FileTransformer sharedFileTransformer;
}

File: android/src/main/java/com/margelo/nitro/fs2/RNFSFileTransformer.kt
package com.margelo.nitro.fs2

object RNFSFileTransformer {
  interface FileTransformer {
    fun onWriteFile(data: ByteArray): ByteArray
    fun onReadFile(
            data: ByteArray
    ): ByteArray // Retained as per original, though usage not seen in manager
  }

  @JvmStatic // If it needs to be accessed from Java as a static field
  var sharedFileTransformer: FileTransformer? = null
}
